pmd-xml/src/main/resources/category/xml/bestpractices.xml

```diff
@@ -8,4 +8,31 @@
     <description>
 Rules which enforce generally accepted best practices.
     </description>
+    
+    <rule name="MissingEncoding"
+      language="xml"
+      message="Set an explicit XML encoding in the XML declaration to ensure proper parsing"
+      class="net.sourceforge.pmd.lang.rule.XPathRule">
+       <description>
+          When the character encoding is missing from the XML declaration,
+          the parser may produce garbled text.
+    
+          This is completely dependent on how the parser is set up
+          and the content of the XML file, so it may be hard to reproduce.
+    
+          Providing an explicit encoding ensures  accurate and consistent
+          parsing.
+       </description>
+       <priority>3</priority>
+       <properties>
+          <property name="version" value="3.1"/>
+          <property name="xpath">
+             <value>
+    <![CDATA[
+    //document[@XmlEncoding = ""]
```

Comment:
```suggestion
    /document[@XmlEncoding = ""]
```

This otherwise looks ok to me, but I think ideally we would expose this through an XPath function, no an attribute. Otherwise it is only accessible through XPathRule, while DomXPathRule is supposed to be preferred for the XML module. `/*[pmd-xml:fileEncoding() = ""]` would be available in both.